src/handlers/canonicalize_issue_links.rs

```diff
@@ -0,0 +1,112 @@
+//! This handler is used to canonicalize linked GitHub issues into their long form
+//! so that when pulling subtree into the main repository we don't accidentaly
+//! close issues in the wrong repository.
+//!
+//! Example: `Fixes #123` (in rust-lang/clippy) would now become `Fixes rust-lang/clippy#123`
+
+use std::borrow::Cow;
+use std::sync::LazyLock;
+
+use regex::Regex;
+
+use crate::{
+    config::CanonicalizeIssueLinksConfig,
+    github::{IssuesAction, IssuesEvent},
+    handlers::Context,
+};
+
+// Taken from https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue?quot#linking-a-pull-request-to-an-issue-using-a-keyword
+static LINKED_RE: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new("(?i)(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved) +(#[0-9]+)")
```

Comment:
```suggestion
    Regex::new("(?i)(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)(:? +)(#[0-9]+)")
```
It is common to write something like `Fixes: #1`, so we should count with the colon, I think. Also, it would be nice to capture everything in between and then put it back when replacing the issue number, to avoid making further changes, even if they only affect whitespace.

![image](https://github.com/user-attachments/assets/d7edb88b-263d-443b-87dc-e6c6d71ede6f)

Note: after this change, the replace string should become `let replace_by = format!("${{1}}${{2}}{full_repo_name}${{3}}");`

Reply:
Good call about the colon, I forgot it was accepted, fixed.

As suggested I also re-added the exact spaces.

And to improve the readability I used named-group instead of unnamed ones.